microcosm_flask/errors.py

```diff
@@ -0,0 +1,111 @@
+"""
+Generalized error handling.
+
+"""
+from logging import getLogger
+
+from flask import jsonify
+from werkzeug.exceptions import default_exceptions
+
+
+error_logger = getLogger("errors")
+
+
+def extract_status_code(error):
+    """
+    Extract an error code from a message.
+
+    """
+    try:
+        return error.code
+    except AttributeError:
+        try:
+            return error.status_code
+        except AttributeError:
+            try:
+                return error.errno
```

Comment:
Note that this would cause *client* errors from `Requests` to be propagated outward. That will often not be the correct behavior semantically, but it's an ok default.